backend/bytecode.proto

```diff
@@ -0,0 +1,206 @@
+syntax = "proto2";
+
+// x : \tau (used in methods and method types)
+message MethodArgument {
+    required string variable = 1;
+    required Type type = 2;
+}
+
+// Variable declarations are either val or var
+enum VariableDeclarationType {
+    VAL = 1;
+    VAR = 2;
+}
+
+// type L = T
+message TypeDeclaration {
+    required string name = 1;
+    required TypeDesc typeDesc = 2;
+}
+
+// b
+message Bytecode {
+    // v
+    message Version {
+        // Magic number? Having one doesn't make sense if we're using protobuf
+        // since it can do validation of bytestreams.
+        required string major = 1;
+        required string minor = 2;
+    }
+    // i
+    message Import {
+        enum ImportType {
+            METADATA = 1;
+            TYPE = 2;
+        }
+        optional ImportType importType = 1;
+        required string platform = 2;
+        // Need to make a global decision here: In the bytecode should we store
+        // paths as '.' delimited strings or arrays of strings? What about
+        // little 'p' paths used in match expressions, datatags, etc?
+        required string path = 3;
+        required Type type = 4;
+        required string name = 5;
+    }
+    required Version version = 1;
+    required string path = 2;
+    repeated Import imports = 3;
+    required Module module = 4;
+}
+
+// M
+message Module {
+    message ExpressionModule {
```

Comment:
I like the term ValueModule a bit better, even if there is an expression.  The point is that the expression evaluates to a pure value, so we can think about it that way.